source/simulator/freesurface.cc

```diff
@@ -172,17 +210,35 @@ namespace aspect
       VectorTools::interpolate_boundary_values (free_surface_dof_handler, *p,
                                                 ZeroFunction<dim>(dim), mesh_displacement_constraints);
 
-    //Zero out the displacement for the prescribed velocity
+    // Zero out the displacement for the prescribed velocity boundaries
+    // if the boundary is not in the set of tangential mesh boundaries
     for (std::map<types::boundary_id, std::pair<std::string, std::string> >::const_iterator p = sim.parameters.prescribed_velocity_boundary_indicators.begin();
          p != sim.parameters.prescribed_velocity_boundary_indicators.end(); ++p)
-      VectorTools::interpolate_boundary_values (free_surface_dof_handler, p->first,
-                                                ZeroFunction<dim>(dim), mesh_displacement_constraints);
+      {
+        if (tangential_mesh_boundary_indicators.find(p->first) == tangential_mesh_boundary_indicators.end())
+          {
+            VectorTools::interpolate_boundary_values (free_surface_dof_handler, p->first,
+                                                      ZeroFunction<dim>(dim), mesh_displacement_constraints);
+          }
+      }
+
+    // Zero out the displacement for the traction boundaries
+    // if the boundary is not in the set of tangential mesh boundaries
+    for (std::map<types::boundary_id, std::pair<std::string, std::string> >::const_iterator p = sim.parameters.prescribed_traction_boundary_indicators.begin();
```

Comment:
Yes. I think at the time you wrote this code traction boundary conditions were not implemented, and we did not think about putting them in at the time the boundary conditions were merged. It seems like nobody ran into problems so far though :smile:.